### ISSUE TITLE
Wildcard V2: `<Popover />` Manual migration

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#27695]
## Manual migration required to update our code to use the `<Popover />` Wildcard component.

See [Wildcard V2 - Planned work](https://docs.google.com/document/d/1NisbJPiadtt5jQw4vUUYJOr8dD6Urwn5V0mVq2wt6bE/edit#heading=h.g4cw92w3ouhw) for more context

Note: Before doing this, we should check if there is scope for a codemod to automate this migration. If so, the codemod should be implemented and ran before starting work on this issue.

## Implementation

This issue assumes the work to implement the Popover component has been done in https://github.com/sourcegraph/sourcegraph/issues/27694

### Component migration:
1. Migrate usage of `@reach` Popover to Wildcard Popover
1. Migrate usage of `reactstrap` Popover to Wildcard Popover
1. Migrate usage of `DropdownMenu` components from `Reactstrap` to Wildcard Popover. See list below

Note: Some `Dropdown*` component usage will be covered by the `Menu` components, only the following components should be migrated to `Popover`:
- StatusMessagesNavItem.tsx
- FeedbackPrompt.tsx
- SearchHelpDropdownButton.tsx
- StreamingProgressSkippedButton.tsx

### Styles migration:
1. Migrate usage of `dropdown-menu` class to Wildcard Popover
1. Remove `dropdown-menu` Bootstrap dependency - reimplement styles in Popover CSS module

Note: `dropdown-menu` usage is also required for the `Menu` components. We should wait until this migration (https://github.com/sourcegraph/sourcegraph/issues/27701) is finished before migrating away from global styles.

## Acceptance criteria
- [ ] All incorrect usage of the previous pattern has been updated to use the Wildcard component

**Comment by umpox** [comment]
Needs scoping

**Comment by umpox** [comment]
Specific components that I have noticed that should not be covered by `<Menu />`: https://github.com/sourcegraph/sourcegraph/issues/27701

- StatusMessagesNavItem.tsx

**Comment by gitstart-sourcegraph** [comment]
Reopen this issue for tracking remaining works in `Styles migration` section

**Comment by gitstart-sourcegraph** [comment]
Hi @valerybugakov @umpox 
An update on the state `Style Migration` requirement of this issue.
Migration of the `dropdown-menu` className has been completed, however since some reactstrap's component (`Dropdown*`) uses the className internally, we cannot safely remove `dropdown` style.
i.e this issue is dependent on #32820

**Comment by valerybugakov** [comment]
> reimplement styles in Popover CSS module

Let's omit this part in this ticket. Focus only on this item: "Remove dropdown-menu Bootstrap dependency".

**Comment by valerybugakov** [comment]
The `Popover` migration is completed. The part of removing the Bootstrap styles and migrating them to CSS modules will be a part of another ticket: https://github.com/sourcegraph/sourcegraph/issues/29200. Closing this one.
